ci-support.sh

```diff
@@ -434,7 +434,8 @@ build_docs()
 
 maybe_upload_docs()
 {
-  if test -n "${DOC_UPLOAD_KEY}" && test "$CI_DEFAULT_BRANCH" && test "$CI_COMMIT_REF_NAME" = "$CI_DEFAULT_BRANCH"; then
+  if test -n "${DOC_UPLOAD_KEY}" && test "$CI_DEFAULT_BRANCH" && \
+      test "$CI_COMMIT_BRANCH"  && test "$CI_COMMIT_BRANCH" = "$CI_DEFAULT_BRANCH"; then
```

Comment:
Didn't test it, but if I understand the [gitlab docs](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#predefined-variables-reference) this should be fine.